circuit/src/ecc/same_curve/mod.rs

```diff
@@ -848,6 +849,147 @@ where
+	ladder: EccUnreducedLadderConfig,
+	pub(crate) add: EccAddConfig,
+	double: EccDoubleConfig,
+	table_select: EccTableSelectConfig,
+	bits2num: Selector,
+}
+
+impl EccBatchedMulScalarConfig {
+	/// Construct a new config
+	pub fn new(
+		ladder: EccUnreducedLadderConfig, add: EccAddConfig, double: EccDoubleConfig,
+		table_select: EccTableSelectConfig, bits2num: Selector,
+	) -> Self {
+		Self { ladder, add, double, table_select, bits2num }
+	}
+}
+
+/// Chipset structure for the EccBatchedMulScalar.
+pub struct EccBatchedMulScalarChipset<
+	C: CurveAffine,
+	const NUM_LIMBS: usize,
+	const NUM_BITS: usize,
+	P,
+> where
+	P: RnsParams<C::Base, C::Scalar, NUM_LIMBS, NUM_BITS>,
+	C::Base: FieldExt,
+	C::Scalar: FieldExt,
+{
+	// Assigned points
+	p: [AssignedPoint<C, NUM_LIMBS, NUM_BITS, P>; NUM_LIMBS],
+	// Assigned scalar values
+	scalar: [AssignedCell<C::Scalar, C::Scalar>; NUM_LIMBS],
+	// AuxInitial (to_add)
+	aux_init: AssignedPoint<C, NUM_LIMBS, NUM_BITS, P>,
+	// AuxFinish (to_sub)
+	aux_fin: AssignedPoint<C, NUM_LIMBS, NUM_BITS, P>,
+}
+
+impl<C: CurveAffine, const NUM_LIMBS: usize, const NUM_BITS: usize, P>
+	EccBatchedMulScalarChipset<C, NUM_LIMBS, NUM_BITS, P>
+where
+	P: RnsParams<C::Base, C::Scalar, NUM_LIMBS, NUM_BITS>,
+	C::Base: FieldExt,
+	C::Scalar: FieldExt,
+{
+	/// Creates a new ecc mul chipset.
+	pub fn new(
+		p: [AssignedPoint<C, NUM_LIMBS, NUM_BITS, P>; NUM_LIMBS],
+		scalar: [AssignedCell<C::Scalar, C::Scalar>; NUM_LIMBS],
+		aux_init: AssignedPoint<C, NUM_LIMBS, NUM_BITS, P>,
+		aux_fin: AssignedPoint<C, NUM_LIMBS, NUM_BITS, P>,
+	) -> Self {
+		Self { p, scalar, aux_init, aux_fin }
+	}
+}
+
+impl<C: CurveAffine, const NUM_LIMBS: usize, const NUM_BITS: usize, P> Chipset<C::Scalar>
+	for EccBatchedMulScalarChipset<C, NUM_LIMBS, NUM_BITS, P>
+where
+	P: RnsParams<C::Base, C::Scalar, NUM_LIMBS, NUM_BITS>,
+	C::Base: FieldExt,
+	C::Scalar: FieldExt,
+{
+	type Config = EccBatchedMulScalarConfig;
+	type Output = Vec<AssignedPoint<C, NUM_LIMBS, NUM_BITS, P>>;
+
+	/// Synthesize the circuit.
+	fn synthesize(
+		self, common: &CommonConfig, config: &Self::Config, mut layouter: impl Layouter<C::Scalar>,
+	) -> Result<Self::Output, Error> {
+		let sliding_window_size = 2;
+
+		let mut aux_init = self.aux_init.clone();
+		let mut aux_inits: Vec<AssignedPoint<C, NUM_LIMBS, NUM_BITS, P>> = Vec::new();
+		for _ in 0..self.scalar.len() {
+			aux_inits.push(aux_init.clone());
+			let double_chip = EccDoubleChipset::new(aux_init);
+			aux_init = double_chip.synthesize(
+				common,
+				&config.double,
+				layouter.namespace(|| "init_double"),
+			)?;
+		}
+
+		let mut aux_fin = self.aux_fin.clone();
+		let mut aux_fins: Vec<AssignedPoint<C, NUM_LIMBS, NUM_BITS, P>> = Vec::new();
+		for _ in 0..self.scalar.len() {
+			aux_fins.push(aux_fin.clone());
+			let double_chip = EccDoubleChipset::new(aux_fin);
+			aux_fin = double_chip.synthesize(
+				common,
+				&config.double,
+				layouter.namespace(|| "fin_double"),
+			)?;
+		}
```

Comment:
Do we need to do this inside the chip. Looks like they are deterministic/same every time. Can we put them inside fixed columns?

Reply:
You are right. I will send them from outside as we did [here](https://github.com/eigen-trust/protocol/blob/master/circuit/src/ecc/same_curve/mod.rs#L950-L1063). I will calculate these values with given `pair length` and send it to the circuit.

Reply:
I changed our `aux_init` and `aux_fin` [calculations](https://github.com/eigen-trust/protocol/blob/batched_mul_chipset/circuit/src/ecc/same_curve/mod.rs#L1166-L1246) in test cases, so we are calculating all of the operations outside of the chip.